runtime/doc/msgpack_rpc.txt

```diff
@@ -387,10 +392,32 @@ of update.
 ["update_menu"]
 	The menu mappings changed.
 
-["mode_change", mode]
-	The mode changed. Currently sent when "insert", "replace", "cmdline" and
-	"normal" modes are entered. A client could for instance change the cursor
-	shape.
+["mode_info_set", cursor_style_enabled, mode_info]
+`mode_info` is an array containing information about various modes, as later
+will be indicated with the `mode_change` update. Currently this primarily
+concerns cursor styles to be set in the different modes. The keys are
+
+	`cursor_shape`:	One of "block", "horizontal" or "vertical"
+	`cell_percentage`:Percentage of the cell taken by the cursor.
+	`blinkwait`, `blinkon`, `blinkoff`: see |cursor-blinking|.
+	`hl_id`:	Highlight group for the cursor.
+	`hl_lm`:	Highlight group for the cursor when langmap is active.
+	`short_name`:	Short name for the mode, as used in 'guicursor'
+			option.
+	`name`:		Full name for the mode.
+	`mouse_shape`:	(To be implemented.)
```

Comment:
We should mention mode_idx here.

    mode_idx: Opaque id received from the `mode_change` event.